guide/src/python_from_rust.md

```diff
@@ -236,6 +236,75 @@ def leaky_relu(x, slope=0.01):
 # }
 ```
 
+### Want to embed Python in Rust with exposed modules?
+
+Internally Python maintains the `sys.modules` dict.
+An import in Python is simply a lookup into this dict.
+
+It is necessary to insert the new module into the dict
+before it is exposed to Python. There are two ways of
+doing this.
```

Comment:
One more problem here: the `append_to_inittab` way does *not* add the module into `sys.modules`. It just makes it findable in one of the locations that are tried when it is *not found* there.

Reply:
Is this an implementation detail that might change between versions?

Reply:
I don't think so. `append_to_inittab` makes the module known without importing it (i.e. executing the module init function). This is a separate step and should not be mixed with the import itself.